packages/tools/src/tools/segmentation/BrushTool.ts

```diff
@@ -153,8 +157,19 @@ class BrushTool extends LabelmapBaseTool {
     }
   ) {
     super(toolProps, defaultToolProps);
+
+    // Add event listener for labelmap undo
+    this._labelmapUndoHandler = this._labelmapUndoHandler.bind(this);
+    eventTarget.addEventListener(
+      Events.LABELMAP_UNDO,
+      this._labelmapUndoHandler
+    );
   }
 
+  private _labelmapUndoHandler = () => {
+    this._previewData.isDrag = true;
```

Comment:
It should compute this based on the event data - so that if the isDrag had been true, it will be again, and if it was false previously it will stay false.  That way it works for undo and redo regardless of whether the before/after values were with drag or not.